src/main/java/frc/robot/subsystems/amp/AmpIOSim.java

```diff
@@ -0,0 +1,140 @@
+package frc.robot.subsystems.amp;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import frc.robot.Constants;
+
+public class AmpIOSim implements AmpIO {
+  private final SingleJointedArmSim sim =
+      new SingleJointedArmSim(
+          DCMotor.getKrakenX60(1),
+          67.5, // gearing
+          0.192383865, // MOI
+          0.3, // arm length
+          Units.degreesToRadians(0), // min angle -- floor
+          Units.degreesToRadians(180), // max angle -- hard stop
+          false,
+          Units.degreesToRadians(0));
+  private final PIDController pid = new PIDController(0.0, 0.0, 0.0);
+
+  private FlywheelSim spinnerSim = new FlywheelSim(DCMotor.getFalcon500(2), 1.5, 0.004);
```

Comment:
You should probably use a dc motor sim for the spinners because they use rollers not fly wheels. They also use a kraken instead of falcon i think